src/simplnx/Utilities/Math/GeometryMath.cpp

```diff
@@ -54,67 +54,67 @@ BoundingBox3Df nx::core::GeometryMath::FindBoundingBoxOfVertices(INodeGeometry0D
   return {ll, ur}; // should be valid
 }
 
-BoundingBox3Df nx::core::GeometryMath::FindBoundingBoxOfFace(const TriangleGeom& faces, int32 faceId)
+BoundingBox3Df nx::core::GeometryMath::FindBoundingBoxOfFace(const detail::GeometryStoreCache& cache, const nx::core::TriangleGeom& triangleGeom, int32 faceId, std::vector<usize>& verts)
 {
-  std::array<Point3Df, 3> points;
-  faces.getFaceCoordinates(faceId, points);
+  // Unavoidable branch to verify integrity of subsequent function
+  // specifically because this an exposed function, however, the
+  // CPU should quickly be able to predict once it has run a few
+  // iterations of the function making it relatively lightweight
+  if(verts.size() != cache.NumVertsPerFace)
+  {
+    verts = std::vector<usize>(cache.NumVertsPerFace);
+  }
+  std::array<Point3Df, 3> points = GeometryMath::detail::GetFaceCoordinates<float32>(cache, faceId, verts);
 
   Point3Df ll = points[0];
   Point3Df ur = points[0];
 
-  if(points[1][0] < ll[0])
-  {
-    ll[0] = points[1][0];
-  }
-  if(points[1][0] > ur[0])
-  {
-    ur[0] = points[1][0];
-  }
-  if(points[1][1] < ll[1])
-  {
-    ll[1] = points[1][1];
-  }
-  if(points[1][1] > ur[1])
-  {
-    ur[1] = points[1][1];
-  }
-  if(points[1][2] < ll[2])
-  {
-    ll[2] = points[1][2];
-  }
-  if(points[1][2] > ur[2])
-  {
-    ur[2] = points[1][2];
-  }
-  if(points[2][0] < ll[0])
-  {
-    ll[0] = points[2][0];
-  }
-  if(points[2][0] > ur[0])
-  {
-    ur[0] = points[2][0];
-  }
-  if(points[2][1] < ll[1])
-  {
-    ll[1] = points[2][1];
-  }
-  if(points[2][1] > ur[1])
-  {
-    ur[1] = points[2][1];
-  }
-  if(points[2][2] < ll[2])
-  {
-    ll[2] = points[2][2];
-  }
-  if(points[2][2] > ur[2])
-  {
-    ur[2] = points[2][2];
-  }
+  // Avoid Branch Prediction misses by exclusively doing multiplication operations
+  // C++ (§4.5/4):
+  // An r-value of type bool can be converted to an r-value of type int,
+  // with false becoming zero and true becoming one.
+  //
+  // The standard guarantees the same for float
+  bool swap = points[1][0] < ll[0];
```

Comment:
I'm not sure that this actually contributes to the speed up. Switching between the two versions gives similar times, and looking at the assembly this version actually generates more assembly than the other.

https://godbolt.org/z/fvYcxMfaf